thriftbp/client_pool_test.go

```diff
@@ -190,6 +193,49 @@ func TestBehaviorWithNetworkIssues(t *testing.T) {
 	}
 }
 
+type thriftHostnameHandler struct {
+	server baseplate.Server
+}
+
+func (thriftHostnameHandler) IsHealthy(ctx context.Context, _ *baseplatethrift.IsHealthyRequest) (r bool, err error) {
+	value, ok := thrift.GetHeader(ctx, thriftbp.ThriftHostnameHeader)
+	if !ok {
+		return false, errors.New("did not find the thrift header")
```

Comment:
I would slightly prefer to add `tb testing.TB` into `thriftHostnameHandler` struct and use `tb.Error()` instead, but this also works.